### PR TITLE
Added IconHeading component

### PR DESCRIPTION
**Pull request by joyheron**
Added IconHeading component with different icons for each major section on the main page.

Also moved the glossary onto an FAQ page and add FAQ as a section.

### CODE REVIEW

**Comment by joyheron** [comment]
Adds some design by way of icons for each major section on the main page.

Screen Shot 2020-02-25 at 21 06 46

File: views/components/icon-heading/_icon-heading.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  .badge {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-basis: 4rem;
+    width: 4rem;
+    height: 4rem;
+    background-color: #D7FDFE;
+    border-radius: 50%;
+    margin: -0.75rem 0.25rem -0.75rem -1rem;
```
Comment (FND) [inline views/components/icon-heading/_icon-heading.scss]: I looks to me (not an authority) like we could do with a little more spacing on the right here.

File: content/index.md [hunk 2]
```diff
@@ -36,24 +36,35 @@ that are [reusable across multiple languages and frameworks](#portability).
 
 This `Card` component is [implemented as a JavaScript function](#components).
 
-
-Stateless HTML Rendering
-------------------------
+<div class="icon-heading">
+    <div class="badge">
+        <svg class="icon" role="presentation">
+            <use href="#icon-server-heart" xlink:href="#icon-server-heart"></use>
```
Comment (FND) [inline content/index.md]: AFAIK `xlink:href` is no longer needed:
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/href

I might be wrong though.

File: content/index.md [hunk 3]
```diff
@@ -82,38 +99,30 @@ Note that this also means authors need basic familiarity with JavaScript and a c
...
 
-Getting Started <span id="getting-started"></span>
----------------
+<div class="icon-heading">
+    <div class="badge">
+        <svg class="icon" role="presentation">
+            <use href="#icon-getting-started" xlink:href="#icon-getting-started"></use>
+        </svg>
+    </div>
+    <h2 id="components">Getting Started</h2>
```
Comment (FND) [inline content/index.md]:
```suggestion
    <h2 id="getting-started">Getting Started</h2>
```

File: content/index.md [hunk 4]
```diff
@@ -82,38 +99,30 @@ Note that this also means authors need basic familiarity with JavaScript and a c
...
-<dd>
-
-Sometimes also known as "isomorphic JavaScript", this means using the same components (macros) on both the server and the client.
+<div class="icon-heading">
+    <div class="badge">
+        <svg class="icon" role="presentation">
+            <use href="#icon-faq" xlink:href="#icon-faq"></use>
+        </svg>
+    </div>
+    <h2 id="components">Frequently Asked Questions</h2>
```
Comment (FND) [inline content/index.md]:
```suggestion
    <h2>Frequently Asked Questions</h2>
```

(Though it'd be nice if all sections had an ID for deep linking, even if it's not currently exposed via the UI.)

**Review by FND — COMMENTED** [review]
This makes a big difference - good choice of icons too!

Just a few observations. Also, did you forget to commit `faq.{md,html}`?

File: views/components/icon-heading/_icon-heading.scss (hunk 1 above)
Comment (joyheron) [reply views/components/icon-heading/_icon-heading.scss]: You're probably right. 🤔 I'll play around with the right spacing a few more times.

**Comment by joyheron** [comment]
Thanks for the feedback! 

I did forget to commit the `faq.{md,html}` files, so I've added a second commit.

**Review by FND — APPROVED** [review]
Now that the FAQ files are there, I think this is good to merge: It's clearly an improvement and tweaks can build upon this later on.

Speaking of the FAQ: It'd help to highlight the linked term (e.g. `:target { background: yellow }`; cf. default styling for `<mark>`).

**Comment by joyheron** [comment]
I've added styling for the linked term (with a fat border bottom). I didn't want to do yellow because we already have way too many colors already. I'm sure that needs to be tweaked, but it might be ok for now.

**Comment by FND** [comment]
Thank you. :shipit:

**Pull request merged**